internal/rbd/errors.go

```diff
@@ -45,4 +45,24 @@ var (
 	// ErrLastSyncTimeNotFound is returned when last sync time is not found for
 	// the image.
 	ErrLastSyncTimeNotFound = errors.New("last sync time not found")
+	// ErrFailedPrecondition is returned when operation is rejected because the system is not in a state
+	// required for the operation's execution.
+	ErrFailedPrecondition = errors.New("system is not in a state required for the operation's execution")
+	// ErrServiceUnavailable is returned when the service is currently unavailable.
+	// This is a most likely a transient condition and may be corrected
+	// by retrying with a backoff. Note that it is not always safe to retry
+	// non-idempotent operations.
+	ErrServiceUnavailable = errors.New("service is currently unavailable")
```

Comment:
Please adapt this generic definition to the context.

Just `ErrUnavailable` is fine. 
In this context
```
// If we issued a resync, return a non-final error as image needs to be recreated
		// locally. Caller retries till RBD syncs an initial version of the image to
		// report its status in the resync request.
		return status.Error(codes.Unavailable, "awaiting initial resync due to split brain")
```

Something like Send this err to trigger a retry from the caller side.